karapace/protobuf/io.py

```diff
@@ -92,24 +91,26 @@ def read_data(writer_schema: ProtobufSchema, reader_schema: ProtobufSchema, bio:
 class ProtobufDatumReader:
     """Deserialize Protobuf-encoded data into a Python data structure."""
 
-    def __init__(self, writer_schema: ProtobufSchema = None, reader_schema: ProtobufSchema = None) -> None:
+    def __init__(self, config: dict, writer_schema: ProtobufSchema = None, reader_schema: ProtobufSchema = None) -> None:
         """As defined in the Protobuf specification, we call the schema encoded
         in the data the "writer's schema", and the schema expected by the
         reader the "reader's schema".
         """
+        self.config = config
         self._writer_schema = writer_schema
         self._reader_schema = reader_schema
 
     def read(self, bio: BytesIO) -> None:
         if self._reader_schema is None:
             self._reader_schema = self._writer_schema
-        return protobuf_to_dict(read_data(self._writer_schema, self._reader_schema, bio), True)
+        return protobuf_to_dict(read_data(self.config, self._writer_schema, self._reader_schema, bio), True)
 
 
 class ProtobufDatumWriter:
     """ProtobufDatumWriter for generic python objects."""
 
-    def __init__(self, writer_schema: ProtobufSchema = None):
+    def __init__(self, config: dict, writer_schema: ProtobufSchema = None):
```

Comment:
nit pick: Please fix typing when touching code:

- config type should be `Config` from `karapace.config`
- `writer_schema` should have an explicit `Optional` (configuration dependent, module is probably does not have the setting set)
- initializer should have explicit return type `-> None` (good practice, not required by mypy)

```suggestion
    def __init__(
        self,
        config: Config,
        writer_schema: Optional[ProtobufSchema] = None,
    ) -> None:
```